indexer/src/fastnear_indexer.rs

```diff
@@ -0,0 +1,195 @@
+use std::collections::HashMap;
+use near_indexer::near_primitives::{types::AccountId, views::{ActionView, ExecutionStatusView, ReceiptEnumView}};
+use reqwest::Client;
+use tokio::sync::{mpsc::{Sender, Receiver}, mpsc};
+use tracing::{info, error, debug, trace};
+
+use crate::{errors::Error, rmq_publisher::{get_routing_key, PublishData, PublishOptions, PublishPayload, PublisherContext}, types::{BlockWithTxHashes, IndexerExecutionOutcomeWithReceiptAndTxHash, PartialCandidateData, PartialCandidateDataWithBlockTxHash}};
+
+const FASTNEAR_ENDPOINT: &str = "https://testnet.neardata.xyz/v0/last_block/final";
+
+#[derive(Debug)]
+pub struct FastNearIndexer {
+    client: Client,
+    addresses_to_rollup_ids: HashMap<AccountId, u32>,
+}
+
+impl FastNearIndexer {
+    pub(crate) fn new(addresses_to_rollup_ids: HashMap<AccountId, u32>) -> Self {
+        debug!(target: "fastnear_indexer", "Creating new FastNearIndexer");
+        Self {
+            client: Client::new(),
+            addresses_to_rollup_ids,
+        }
+    }
+
+    pub fn run(&self) -> Receiver<PublishData> {
+        info!(target: "fastnear_indexer", "Starting FastNearIndexer");
+        let block_receiver = self.stream_latest_blocks();
+        let (publish_sender, publish_receiver) = mpsc::channel(100);
```

Comment:
might wanna configure these channel bounds in a config at some point

Reply:
Will do!